rise/static/main.js

```diff
@@ -13,82 +13,104 @@ define([
+   *
+   * it returns a promise that can be then'ed once the config is loaded
+   *
+   * setup waits for the config to be loaded before it actually enables keyboard shortcuts 
+   * and other menu items; so this means that the bulk of the code can assume that the config
+   * is already loaded and does not need to worry about using promises, or 
+   * waiting for any asyncronous code to complete
    */
-  function configSlides() {
 
-    var default_config = {
-      controls: true,
-      progress: true,
-      history: true,
-      width: "100%",
-      height: "100%",
-      margin: 0.1,
-      minScale: 1.0, // we need this for codemirror to work right
-      theme: 'simple',
-      transition: 'linear',
-      slideNumber: true,
-      /* describe where to start slideshow
-       * can be either:
-       * 'beginning' : start on first slide 
-       * 'selected'  : start on current slide
-       * xxx: could be useful to add more policies
-       * like alowing to start on a hard-wired slide number
-       */
+  var complete_config = {};
+
+  function loadConfig() {
+
+    // see rise.yaml for more details
+    var hardwired_config = {
+
+      // behaviour
+      autolaunch: false,
       start_slideshow_at: 'selected',
-      /* describe how to select cells when new contents
-       * is displayed (new slide or new fragment)
-       * can be either:
-       * 'none' - no autoselect
-       * 'code' - auto-select first code cell if any
-       * 'first' - select first cell - should be
-       * considered an experimental attempt, does
-       * not seem very helpful in real life
-       */
       auto_select: 'code',
-      /* if auto_select is not 'none', this boolean
-       * says if selection focuses on the current fragment
-       * or considers the whole slide
-       */
       auto_select_fragment: true,
+
+      // aspect
+      header: undefined,
+      footer: undefined,
+      backimage: undefined,
+      overlay: undefined,
+
+      // UI
+      toolbar_icon: 'fa-bar-chart',
+      shortcuts: {
+        'slideshow' : 'alt-r',
+        'toggle-slide': 'shift-i',
+        'toggle-subslide': 'shift-u',
+        'toggle-fragment': 'shift-f',
+        // unassigned by default
+        'toggle-note': '',
+        'toggle-skip': '',
+      },
+
+      // reveal native settings passed as-is
+      theme: 'simple',
+      transition: 'linear',
+      // xxx there might be a need to tweak this one when set
+      // by the configurator, as e.g. 'false' or 'true' will result
+      // in a string and not a boolean
+      slideNumber: true,
+      width: "100%",
+      height: "100%",
+      controls: true,
+      progress: true,
+      history: true,
       scroll: false,
       center: true,
-      autolaunch: false,
+      margin: 0.1,
+      minScale: 1.0, // we need this for codemirror to work right
     };
 
-    var config_section = new configmod.ConfigSection(
-      'livereveal',
```

Comment:
See my comment above about `livereveal.json`.

Reply:
My general impression, ever since I started using RISE, is that the 2 names livereveal and rise are confusing to many.
This is why in this release I support both names.

Once I get a better understanding of this .json thing I can reproduce here the same policy as with the notebook metadata, i.e. honor both sources of settings; in practice that would mean use settings from rise.json, then from livereveal.json if still not specified.  
But I need to understand the json thing better, of course.

Reply:
>is that the 2 names livereveal and rise are confusing to many.

I agree, I think it is OK to support both names for compatibility reasons but we should use rise instead of livereveal going forward.

> in practice that would mean use settings from rise.json, then from livereveal.json if still not specified.

Yep.